Dependencies/xr/Source/OpenXR/XR.cpp

```diff
@@ -1480,6 +1480,15 @@ namespace xr
                     {
                         // Only signal that gamepad data is available if the actions were available
                         InputSources[idx].GamepadTrackedThisFrame = true;
+ 
+                        // map the openxr values to populate other states of a button and axes that webxr expects
+                        gamepadObject.Buttons[controllerInfo.TRIGGER_BUTTON].Pressed = (gamepadObject.Buttons[controllerInfo.TRIGGER_BUTTON].Value == 1);
+                        gamepadObject.Buttons[controllerInfo.TRIGGER_BUTTON].Touched = (gamepadObject.Buttons[controllerInfo.TRIGGER_BUTTON].Value > 0);
+                        gamepadObject.Buttons[controllerInfo.SQUEEZE_BUTTON].Value = (gamepadObject.Buttons[controllerInfo.SQUEEZE_BUTTON].Pressed);
+                        gamepadObject.Buttons[controllerInfo.SQUEEZE_BUTTON].Touched = (gamepadObject.Buttons[controllerInfo.SQUEEZE_BUTTON].Pressed);
+                        gamepadObject.Buttons[controllerInfo.TRACKPAD_BUTTON].Value = (gamepadObject.Buttons[controllerInfo.TRACKPAD_BUTTON].Pressed);
+                        gamepadObject.Buttons[controllerInfo.THUMBSTICK_BUTTON].Value = (gamepadObject.Buttons[controllerInfo.THUMBSTICK_BUTTON].Pressed);
+                        gamepadObject.Buttons[controllerInfo.THUMBSTICK_BUTTON].Touched = (gamepadObject.Axes[controllerInfo.THUMBSTICK_X_AXIS] != 0 || gamepadObject.Axes[controllerInfo.THUMBSTICK_Y_AXIS] != 0);
```

Comment:
This one is interesting. It makes sense at a glance, but I wonder if it's what the end receivers of input are expecting. Do you have a documentation link for this?

Reply:
Yes, I found it here: https://github.com/immersive-web/webxr-input-profiles/tree/master/packages/motion-controllers#thumbstick-and-touchpad-components